pkg/controllers/default_resources.go

```diff
@@ -353,36 +353,25 @@ func ScaffoldAWSNodePoolSpec(hyd *hypdeployment.HypershiftDeployment, infraOut *
 	}
 }
 
-func ScaffoldNodePoolSpec(hyd *hypdeployment.HypershiftDeployment) {
+func ScaffoldNodePoolSpec(hyd *hypdeployment.HypershiftDeployment, infraOut *aws.CreateInfraOutput) {
+	if len(hyd.Spec.NodePools) == 0 {
+		hyd.Spec.NodePools = []*hypdeployment.HypershiftNodePools{}
 
-	replicas := int32(2)
+		if infraOut == nil {
+			hyd.Spec.NodePools = append(hyd.Spec.NodePools, getNodepoolSpec(hyd.Name, hyd.Name))
+		} else {
+			for _, zone := range infraOut.Zones {
+				nodePoolSpecName := hyd.Name
 
-	if len(hyd.Spec.NodePools) == 0 {
-		hyd.Spec.NodePools = []*hypdeployment.HypershiftNodePools{
-			{
-				Name: hyd.Name,
-				Spec: hyp.NodePoolSpec{
-					ClusterName: hyd.Name,
-					Management: hyp.NodePoolManagement{
-						AutoRepair: false,
-						Replace: &hyp.ReplaceUpgrade{
-							RollingUpdate: &hyp.RollingUpdate{
-								MaxSurge:       &intstr.IntOrString{IntVal: 1},
-								MaxUnavailable: &intstr.IntOrString{IntVal: 0},
-							},
-							Strategy: hyp.UpgradeStrategyRollingUpdate,
-						},
-						UpgradeType: hyp.UpgradeTypeReplace,
-					},
-					Replicas: &replicas,
-					Platform: hyp.NodePoolPlatform{
-						Type: hyp.NonePlatform,
-					},
-					Release: hyp.Release{
-						Image: getReleaseImagePullSpec(), //.DownloadURL,,
-					},
-				},
-			},
+				if len(infraOut.Zones) > 1 {
+					// If there are multiple zones, name node pools differently
+					nodePoolSpecName = hyd.Name + "-" + zone.Name
+				}
+
+				nodePoolSpec := getNodepoolSpec(nodePoolSpecName, hyd.Name)
```

Comment:
If we are doing Zones. infraOut.Zones > 1 should we limit replicas to 1 instead of 2?  So one node per nodePool?  Does hypershift cli leave the default 2??

Reply:
Good point. Let me check

Reply:
@jnpacker Hypershift does not have any default node pool replica. I created without specifying it and it set the replica count to 0. I think it is our choice on the number of replicas. 1?

Reply:
```--node-pool-replicas int32                   If >-1, create a default NodePool with this many replicas```
This means it take 0 as well and it appears that 0 is the default

Reply:
Why did we decide to set it to 2 by default?

Reply:
That was the CLi default